diff --git a/matron/wscript b/matron/wscript
index 16472ea..e992648 100644
--- a/matron/wscript
+++ b/matron/wscript
@@ -86,7 +86,9 @@ def build(bld):
         matron_libs += ['stdc++']
         matron_use += ['LIBLINK_C']
 
-    matron_cflags=['-O3', '-Wall', '-std=c11', '-mfpu=neon']
+    matron_cflags=['-O3', '-Wall', '-std=c11']
+    if not bld.env.NORNS_DESKTOP:
+        matron_cflags += ['-mfpu=neon']
 
     if bld.env.NORNS_RELEASE:
         matron_cflags += [
