--- a/matron/src/weaver.c
+++ b/matron/src/weaver.c
@@ -1779,8 +1779,7 @@
     int y = (int)luaL_checkinteger(l, 3) - 1; // convert from 1-base
     int z = (int)luaL_checkinteger(l, 4);     // don't convert value!
     bool rel = lua_toboolean(l, 5);
-    if (emu_enabled() && (x<0 || y<0 || x>=emu_grid_cols() || y>=emu_grid_rows() ||
-        z < (rel ? -127 : 0) || z > (rel ? 127 : 15))) return luaL_error(l,"invalid virtual grid LED coordinate/level");
+    if (emu_enabled() && (x<0 || y<0 || x>=emu_grid_cols() || y>=emu_grid_rows())) return luaL_error(l,"invalid virtual grid LED coordinate/level");
     dev_monome_grid_set_led(md, x, y, z, rel);
     lua_settop(l, 0);
     return 0;
@@ -1811,7 +1810,6 @@
     struct dev_monome *md = lua_touserdata(l, 1);
     int z = (int)luaL_checkinteger(l, 2); // don't convert value!
     bool rel = lua_toboolean(l, 3);
-    if (emu_enabled() && (z < (rel ? -127 : 0) || z > (rel ? 127 : 15))) return luaL_error(l,"invalid virtual grid bulk level");
     dev_monome_all_led(md, z, rel);
     lua_settop(l, 0);
     return 0;
--- a/matron/src/emu_bridge.c
+++ b/matron/src/emu_bridge.c
@@ -128,7 +128,7 @@
         for (int x=0; x<16; ++x) {
             int lx,ly; grid_coordinates(x,y,&lx,&ly);
             int q=((ly>7)<<1)|(lx>7);
-            leds[y*16+x] = grid_connected ? md->data[q][(ly%8)*8+lx%8] : 0;
+            leds[y*16+x] = grid_connected ? (md->data[q][(ly%8)*8+lx%8] & 15) : 0;
         }
     emit(2, md->dev.id, leds, sizeof(leds));
     memset(md->dirty,0,sizeof(md->dirty));
